Dan/solution.py

```diff
@@ -0,0 +1,119 @@
+import random
+
+RIGHT = 'r'
+DOWN = 'd'
+
+class Grid:
+    """
+    I created a class so I could reference the grid without using global variables
+    """
+    def __init__(self,m,n):
+        """
+        creates a playing board for the robot
+        """
+        self.columns = m
+        self.rows = n
+        self.board = makeBoard(m,n)
+
+    def find_good_paths(self):
+        """
+        Returns an array of all the valid paths through the board
+        """
+        return self.robot_step((0,0),[])
```

Comment:
Something that struck me while making mine... Just from solution standpoint, we cannot assume that (0,0) or (m,n) will be empty. However, this question needs to be asked to the interviewer.

Reply:
That's a good point we should ask. Though I think we can either assume (0,0) is not blocked or it doesn't matter because the robot is already there.